framework/uicomponents/qml/MuseScore/UiComponents/TextInputField.qml

```diff
@@ -11,105 +12,105 @@ Rectangle {
 
-        states: [
-            State {
-                name: "NORMAL"
-                when: !clickableArea.containsMouse && !valueInput.focus
+            color: ui.theme.fontColor
 
-                PropertyChanges { target: root; border.color: ui.theme.textFieldColor
-                                                border.width: 1 }
-            },
+            font {
+                family: ui.theme.font.family
+                pointSize: ui.theme.font.pointSize
+            }
 
-            State {
-                name: "HOVERED"
-                when: clickableArea.containsMouse && !valueInput.focus
+            background: Rectangle {
+                color: root.color
+            }
 
-                PropertyChanges { target: root; border.color: ui.theme.strokeColor
-                                                border.width: 1 }
-            },
+            focus: false
+            activeFocusOnPress: false
+            selectByMouse: true
+            selectionColor: Qt.rgba(ui.theme.accentColor.r, ui.theme.accentColor.g, ui.theme.accentColor.b, ui.theme.accentOpacityNormal)
+            selectedTextColor: ui.theme.fontColor
+            visible: !root.isIndeterminate || activeFocus
 
-            State {
-                name: "FOCUSED"
-                when: valueInput.focus && valueInput.selectedText === ""
+            text: root.currentText === undefined ? "" : root.currentText
 
-                PropertyChanges { target: root; border.color: ui.theme.accentColor
-                                                border.width: 1 }
-            },
+            Keys.onPressed: {
+                if (event.key === Qt.Key_Enter || event.key === Qt.Key_Return) {
+                    valueInput.focus = false
+                }
+            }
+
+            onActiveFocusChanged: {
+                if (activeFocus) {
+                    selectAll()
+                } else {
+                    deselect()
+                }
+            }
 
-            State {
-                name: "TEXT_SELECTED"
-                when: valueInput.focus && valueInput.selectedText !== ""
+            onTextChanged: {
+                if (!acceptableInput) {
+                    return
+                }
 
-                PropertyChanges { target: root; border.color: ui.theme.accentColor
-                                                border.width: 1}
+                root.currentTextEdited(text)
             }
-        ]
-    }
+        }
 
-    StyledTextLabel {
-        id: measureUnitsLabel
+        StyledTextLabel {
+            id: measureUnitsLabel
 
-        anchors.left: valueInput.right
-        anchors.leftMargin: 4
-        anchors.verticalCenter: valueInput.verticalCenter
+            Layout.alignment: Qt.AlignVCenter
 
-        color: ui.theme.fontColor
-        visible: !root.isIndeterminate
+            color: ui.theme.fontColor
+            visible: !root.isIndeterminate && Boolean(text)
+        }
+
+        Item {
```

Comment:
похоже на какой-то хак
зачем понадобился?

Reply:
I solved the problem with measureUnitsLabel. This label must be close to the text. But since I have used RowLayout, measureUnitsLabel flew away to the right border. That item allows keeping measureUnitsLabel close to the text